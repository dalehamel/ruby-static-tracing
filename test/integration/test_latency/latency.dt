expensive_operation*:::execute
{
  printf("Attaching 1 probe...\n%s\n\n\n", copyinstr(arg0))
}
