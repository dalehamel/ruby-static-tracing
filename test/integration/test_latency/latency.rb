# frozen_string_literal: true

require 'ruby-static-tracing'
STDOUT.sync = true

class ExpensiveOperation
  def execute
    ([] << 1) * 100_000
  end

  StaticTracing::Tracer::Latency.register(self, :execute)
end

StaticTracing.configure do |config|
  config.add_tracer(StaticTracing::Tracer::Latency)
end

expensive_operation = ExpensiveOperation.new

Signal.trap('USR2') do
  expensive_operation.execute
end

loop do
end
