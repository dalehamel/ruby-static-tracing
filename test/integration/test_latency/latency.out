Attaching 1 probe...
execute
